### ISSUE TITLE
Request for different prefix for DB name in Mongo for LD

### ISSUE THREAD

**Issue by ccsr** [Fiware/context.Orion-LD#376]
Right now Orion v2 is storing in MongoDB as below. So Orion_*Fiware-service*

orion               0.000GB
orion-*Fiware-service1*        0.000GB
orion-*Fiware-service2*       0.000GB

I checked Orion-LD it is also using the same way of naming. I request to do it something like below

Orion**ld**               0.000GB
orion**ld**-*Fiware-service1*        0.000GB
orion**ld**-*Fiware-service2*       0.000GB

If this can be done, then the migration from Orionv2 to Orion-LD can be done in a smooth way by deploying both of the Orion's and gradually migrating to LD.

**Comment by kzangeli** [comment]
Yes, it can be done, shouldn't take me much more than an hour to fix this.

We also have a CLI parameter (`-db <db-name>`) to change the default "basename" of the databases, which right now is "orion" for both protocols.
If we do this we should probably add another CLI - `-lddb <db-name for ngsi-ld>`.

Now, this change would imply a separation of the NGSIv2 entities and the NGSI-LD entities.
An entity created using NGSIv2 will end up in the database "orion" (without considering tenants),
while one created using NGSI-LD would end up in the database "orionld".
A GET /entities using NGSIv2 will only see the entities in the "orion" database. etc.

Orion-LD would be "two brokers in one", only not "mixing" the entities of different APIs (entities, subscriptions, registrations).

I don't really see the use of mixing entities anyway, now that there are so many (and important) changes between NGSIv2 and NGSI-LD, so, for me, this seems like a pretty good idea.

**Comment by jason-fox** [comment]
:+1: from me - I would like to be able to smoothly upgrade from v2 to LD. I already have a script to convert entities to LD format, but obviously this requires a separate database during the conversion phase.

For Orion v2 to Orion LD I would expect to do the following:

1) Read in all entities, convert from v2  to LD - store in LD database
2) Read in all registrations, convert from v2  to LD - store in LD database
3) Read in all subscriptions, convert from v2  to LD - store in LD database.

Now let us assume **something goes wrong**.  I have a ready made rollback strategy - just use the v2 database. 

If the default LD database in `orionld`, but it can be overridden, then if necessary I can rename my LD database(s) back to use the old `orion`  prefix if that helps  my disaster recovery cron-jobs.

**Comment by JuanjoHierro** [comment]
I disagree this is a good idea.  It is not helpful for a gradual migration of existing systems based on NGSIv2 to incorporate developments based on NGSI-LD.

You may have developed a dashboard application using NGSIv2, which monitor a system based on values of attributes of entities in the system.   A very common development in "powered by FIWARE" systems which may have thousands of lines of code using NGSIv2.

You may just consider a map-based dashboard showing entities linked to parking lots, bus stops, location of buses, etc.   Now, a new subsystem dealing with management of bicycles and bicycle slots is developed. And, of course, you want it to be developed using NGSI-LD.   However, you may want to add monitoring of those new entities in the system, showing them in the map.

We have to come with a way values of certain attributes of those new entities (those that are common to the rest and relevant for the visualization of entities in the map) can be read in NGSIv2.

Similarly, you may have entities in the system which were created using NGSIv2 and you may want to read attributes of them using a new application that uses NGSI-LD without having to migrate the apps/modules that were creating them using NGSIv2 (e.g., some IoT agents or other systems which were based on NGSIv2).   And then implement a gradual migration of those context producer/provider apps/modules. 

That is the kind of co-existence we have to support.   I believe the proposed change doesn't help in that direction.  Right the contrary, could make gradual migration more difficult.
